### PR TITLE
Replace RZ_ASM_NOPLUGINS and RZ_BIN_NOPLUGINS with more general RZ_NOPLUGINS

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

All green (since the variable is used for all tests, stuff will explode if something is wrong).

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@ret2libc take a look at this one please, and if you don't have any objections - merge.

**Review by wargio — APPROVED** [review]

File: librz/main/rz-asm.c [hunk 1]
```diff
@@ -203,7 +203,7 @@ static int rasm_show_help(int v) {
 		       " If '-l' value is greater than output length, output is padded with nops\n"
 		       " If the last argument is '-' reads from stdin\n");
 		printf("Environment:\n"
-		       " RZ_ASM_NOPLUGINS  do not load shared plugins (speedup loading)\n"
+		       " RZ_NOPLUGINS  do not load shared plugins (speedup loading)\n"
```
Comment (ret2libc) [inline librz/main/rz-asm.c]: does this display aligned?

**Review by ret2libc — COMMENTED** [review]
Mh, If the others are ok i'm ok, though I would have gone with the opposite solution of keeping one NOPLUGIN for each kind of plugins.

**Comment by Rot127** [comment]
@ret2libc 
> though I would have gone with the opposite solution of keeping one NOPLUGIN for each kind of plugins.

We can do this as well. Though, if we want to have it properly done, we would need to:
- Add checks for `RZ_NOPLUGIN` where ever the other `NOPLUGINS` are.
- Define somewhere what a bin plugin and what an asm plugin is.

Idk, is it worth the trade? We can also open an issue about it for a better plugin loading module.
This is maybe even a nice task for beginners with C knowledge.

Something where each plugin needs to set its category and we have a simply call to enable a certain plugin

```c
load_plugin(plugin) { if (do_not_load_plugin_group(X) return; .... }
```

**Comment by ret2libc** [comment]
I'm not sure I follow you.

> Define somewhere what a bin plugin and what an asm plugin is.

What do you mean? We already have separate bin plugins and separate asm plugins.

**Comment by Rot127** [comment]
> What do you mean? We already have separate bin plugins and separate asm plugins.

Sorry, I was not clear with this.

The flag `BIN/ASM_NOPLUGINS` suggests that only ASM/BIN plugins don't get loaded. But when the flag is checked and is set, no plugin is loaded at all (at the location of the env check).
Also there are no checks for `RZ_HASH_NOPLUGINS` or `RZ_ANALYSIS_NOPLUGINS`, although plugins of this type are loaded as well.

A more general problem is, plugin libraries get loaded via `rz_lib_opendir()`. But this loads always all libraries from this directory (at least this is what the doc string says).
Though the plugin is only usable if the handler was added via `rz_lib_add_handler()`.

This appears inconsistent to me and hard to manage, because there is no single module which does the plugin loading.
`RZ_NOPLUGINS` (or `RZ_XXX_NOPLUGINS`) should be only checked at a single location in the code base IMHO. So if for example ASM plugins should be loaded, a single plugin load handler is called, it checks if `RZ_ASM_NOPLUGINS` or `RZ_NOPLUGINS` is set, loads only the needed libs and adds only the needed handlers.

**Comment by ret2libc** [comment]
I understand now, thanks for the explanation! If so, yes then it makes sense how you did it :) Thanks for the patience :D

**Pull request merged**